libs/knowledge-store/notebooks/support.ipynb

```diff
@@ -0,0 +1,596 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create the RAG Chain\n",
+    "Here, I create two versions of the retriever and chain -- one which uses a depth of 0 (so is equivalent to vector search) and another which uses a depth of 1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_openai import ChatOpenAI\n",
+    "from langchain_core.prompts import ChatPromptTemplate\n",
+    "from langchain_core.runnables import RunnablePassthrough\n",
+    "from langchain_core.output_parsers import StrOutputParser\n",
+    "\n",
+    "llm = ChatOpenAI(model=\"gpt-4o\")\n",
+    "\n",
+    "retriever0 = knowledge_store.as_retriever(search_kwargs={\"depth\": 0})\n",
+    "retriever1 = knowledge_store.as_retriever(search_kwargs={\"depth\": 1})\n",
+    "\n",
+    "template = \"\"\"You are a helpful technical support bot. You should provide complete answers explaining the options the user has available to address their problem. Answer the question based only on the following context:\n",
+    "{context}\n",
+    "\n",
+    "Question: {question}\n",
+    "\"\"\"\n",
+    "prompt = ChatPromptTemplate.from_template(template)\n",
+    "\n",
+    "def format_docs(docs):\n",
+    "    formatted = \"\\n\\n\".join(f\"From {doc.metadata['content_id']}: {doc.page_content}\" for doc in docs)\n",
+    "    return formatted\n",
+    "\n",
+    "rag_chain0 = (\n",
+    "    {\"context\": retriever0 | format_docs, \"question\": RunnablePassthrough()}\n",
+    "    | prompt\n",
+    "    | llm\n",
+    "    | StrOutputParser()\n",
+    ")\n",
+    "\n",
+    "rag_chain1 = (\n",
+    "    {\"context\": retriever1 | format_docs, \"question\": RunnablePassthrough()}\n",
+    "    | prompt\n",
+    "    | llm\n",
+    "    | StrOutputParser()\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import display, Markdown\n",
+    "\n",
+    "# Helper method to render markdown in responses to a chain.\n",
+    "def run_and_render(chain, question):\n",
+    "    result = chain.invoke(question)\n",
+    "    display(Markdown(result))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Check Retrieval Results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Vector Only: https://docs.datastax.com/en/cql/astra/getting-started/vector-search-quickstart.html\n",
+      "Vector Only: https://docs.datastax.com/en/astra-db-serverless/get-started/concepts.html\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Traversing:  https://docs.datastax.com/en/cql/astra/getting-started/vector-search-quickstart.html\n",
```

Comment:
Note that traversing gives us a lot of output. I suspect we would benefit from the MMR retrieval, and would like to add that to this notebook once it goes in.